kv/kvserver: TestLeaseMetricsOnSplitAndTransfer failed

kv/kvserver.TestLeaseMetricsOnSplitAndTransfer [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9781389?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9781389?buildTab=artifacts#/) on master @ [847de4595b62a0c03b14a3d0865cc8222fcf0f61](https://github.com/cockroachdb/cockroach/commits/847de4595b62a0c03b14a3d0865cc8222fcf0f61):


```
=== RUN   TestLeaseMetricsOnSplitAndTransfer
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/3badd79dc476285a5b1104ad1a94c4a5/logTestLeaseMetricsOnSplitAndTransfer2314119911
    test_log_scope.go:79: use -show-logs to present logs inline
    client_replica_test.go:2119: expected 1 lease transfer errors; got 6
    client_replica_test.go:2150: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/3badd79dc476285a5b1104ad1a94c4a5/logTestLeaseMetricsOnSplitAndTransfer2314119911
--- FAIL: TestLeaseMetricsOnSplitAndTransfer (3.08s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #99266 kv/kvserver: TestLeaseMetricsOnSplitAndTransfer failed [C-test-failure O-robot T-kv branch-release-23.1 no-test-failure-activity]
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseMetricsOnSplitAndTransfer.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/client_replica_test.go
+++ b/pkg/kv/kvserver/client_replica_test.go
@@ -2115,8 +2115,11 @@ func TestLeaseMetricsOnSplitAndTransfer(t *testing.T) {
 	if a, e := metrics.LeaseTransferSuccessCount.Count(), int64(1); a != e {
 		t.Errorf("expected %d lease transfer successes; got %d", e, a)
 	}
-	if a, e := metrics.LeaseTransferErrorCount.Count(), int64(1); a != e {
-		t.Errorf("expected %d lease transfer errors; got %d", e, a)
+	// We mostly expect precisely one error, but there's a retry loop in
+	// `AdminTransferLease` that prevents transfers to followers who might need a
+	// snapshot. This can sometimes lead to additional errors being reported.
+	if a := metrics.LeaseTransferErrorCount.Count(); a == 0 {
+		t.Errorf("expected at least one lease transfer errors; got %d", a)
 	}
 
 	// Expire current leases and put a key to the epoch based scratch range to

